ood-portal-generator/lib/ood_portal_generator/dex.rb

```diff
@@ -158,8 +158,29 @@ def client_name
       @config.fetch(:client_name, "OnDemand")
     end
 
+    def default_secret_path
+      File.join(self.class.config_dir, "ondemand.secret")
+    end
+
+    def generate_secret
+      return default_secret_path if (File.exist?(default_secret_path) && ! File.zero?(default_secret_path))
+
+      secret = SecureRandom.uuid
+      File.open(default_secret_path, "w") { |f| f.write("#{secret}\n") }
+      FileUtils.chown(OodPortalGenerator.dex_user, OodPortalGenerator.dex_group, default_secret_path)
+      FileUtils.chmod(0600, default_secret_path)
+      default_secret_path
+    end
+
     def client_secret
-      @config.fetch(:client_secret, Digest::SHA1.hexdigest(client_id))
+      secret = @config.fetch(:client_secret, nil)
+      if secret.nil? && self.class.installed? && enabled?
+        secret = generate_secret
+      end
+      if !secret.nil? && File.exist?(secret)
+        secret = File.read(secret).strip
+      end
+      secret
     end
```

Comment:
If `self.class.installed? && enabled?` is false does it matter whether or not `client_secret` returns nil or a string? If it is irrelevant, then shifting the guard clause up and using the block option for the alternative for secret might be preferable:

```ruby
def client_secret
  return nil unless self.class.installed? && enabled?

  secret = @config.fetch(:client_secret) { generate_secret }
  secret = File.read(secret).strip if File.exist?(secret.to_s)
  secret
end
```

* `File.exist?(nil.to_s) == false`

Reply:
Updated `client_secret` with suggested changes